### DISCUSSION TITLE
Improve integration with flowise

### DISCUSSION THREAD

**Discussion by tsiougkosn** [langfuse/langfuse#953] [Ideas]
From flowise we can pass some attributes to langfuse eg :
```
overrideConfig: {
  analytics: {
      langFuse: {
          chatId: "sampleChatId",
          userId: "sampleUser1",
          sessionId: "sampleSessionId",
      }
  },
  sessionId: "sampleSession"
}
```
If you want to have a reference to the trace in the langfuse, you have to get the traces by session and keep the last (by timestamp). What about to pass the id (trace id) from the flowise to langfuse

**Comment by marcklingen** [comment]
Is there an idea in Flowise that you'd like to reuse in Langfuse as a traceId or do you want to pass your own id via an overrideConfig?

**Reply by tsiougkosn** [reply]
In flowise I suppose they haven't that need. 
But as a user of flowise api, I have my backend to ask the flowise for a question. I later want to have feedback on that answer and keep the feedback / score in the langfuse. So I want to have a map between flowise answer <> langfuse traceId.

Now I can call langfuse api, just after the flowise return the answer, by using the sessionId and getting all the traces (and get the Id of latest timestamp). But that is not bullet proof, so I am thinking for a solid solution.

**Comment by danieldabate** [comment]
Hi @tsiougkosn, I found the same issue today. 

Do you know if there is a new way of solving this and not having to get the last trace id from the session?

**Reply by tsiougkosn** [reply]
Hey Daniel, we used a 'hack' to pass a trace id through the "version" field. FYI, we haven't updated to the latest version, so I'm not sure if this has been fixed in the latest update.

**Reply by marcklingen** [reply]
passing a custom traceId through flowise would be ideal here. This should be a small change if it is not available yet to the langfuse integration in flowise. Can you make the change or raise an issue on the flowise repo?

**Reply by danieldabate** [reply]
I looked at Flowise's code and seems like it should work, but it doesn't. Injecting things like user id and version work just fine.

But I also saw that Flowise is using an old version of Langfuse JS SDK. Something like version 3.3.4.

Do you know if setting custom ids for traces is something that was developed recently?

**Reply by danieldabate** [reply]
I don't think it's related to the Langfuse SDK version, tried updating and it didn't work. I checked and Flowise is using both the `langfuse` and `langfuse-langchain` SDKs. The flow I'm testing uses the `langfuse-langchain` SDK, so probably the limitation is related to that.

I can't find inside the langfuse documentation if there is a way to pass a custom trace id using that SDK. I can only find it with `langfuse` SDK.

Edit: Created an issue on Flowise side https://github.com/FlowiseAI/Flowise/issues/2808

**Reply by danieldabate** [reply]
> Hey Daniel, we used a 'hack' to pass a trace id through the "version" field. FYI, we haven't updated to the latest version, so I'm not sure if this has been fixed in the latest update.

Thanks for the reply! I have some questions about the approach.

What are you setting in the "version" field? A random ID? Are you then mapping on your side that random id with Flowise's message id?

Also, how are you then doing the score? Are you getting all the traces for a session and searching for the one with the version that matches your custom trace id?
